src/nvim/window.c

```diff
@@ -735,6 +728,22 @@ bool parse_float_config(Dictionary config, FloatConfig *out, bool reconf,
                       "'col' option has to be Integer or Float");
         return false;
       }
+    } else if (strequal(key, "width")) {
+      if (val.type == kObjectTypeInteger && val.data.integer >= 0) {
+        out->width = val.data.integer;
+      } else {
+        api_set_error(err, kErrorTypeValidation,
+                      "'width' option has to be a positive Integer");
+        return false;
+      }
+    } else if (strequal(key, "height")) {
+      if (val.type == kObjectTypeInteger && val.data.integer >= 0) {
+        out->height= val.data.integer;
+      } else {
+        api_set_error(err, kErrorTypeValidation,
```

Comment:
the error cases aren't tested. That could help coverage.

The `helpers.expect_err` function makes it easy to test errors.